langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/RegexUtil.java

```diff
@@ -46,7 +50,12 @@ public class RegexUtil {
 
     static final BArrayType GROUPS_ARRAY_TYPE = new BArrayType(GROUPS_AS_SPAN_ARRAY_TYPE);
     static Matcher getMatcher(BRegexpValue regexpVal, BString inputStr) {
-        return getMatcher(regexpVal, inputStr.getValue());
+        try {
+            return getMatcher(regexpVal, inputStr.getValue());
+        } catch (PatternSyntaxException e) {
+            throw BLangExceptionHelper.getRuntimeException(BallerinaErrorReasons.REG_EXP_PARSING_ERROR,
```

Comment:
Let's append the `e.message`

Reply:
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/39444/commits/f9cc62cc815c846f7e99f4711c015f2167bb86eb.